src/adapters/tokio.rs

```diff
@@ -72,6 +72,29 @@ impl<T: tokio::io::AsyncWrite + Unpin + ?Sized> crate::asynch::Write for FromTok
     }
 }
 
+impl<T: tokio::io::AsyncSeek + Unpin + ?Sized> crate::asynch::Seek for FromTokio<T> {
+    type SeekFuture<'a> = impl Future<Output = Result<u64, Self::Error>>
+    where
+        Self: 'a;
+
+    fn seek<'a>(&'a mut self, pos: crate::SeekFrom) -> Self::SeekFuture<'a> {
+        poll_fn::poll_fn(move |cx| {
+            // Note: `start_seek` can return an error if there is another seek in progress.
+            // Therefor it is recommended to call `poll_complete` before any call to `start_seek`.
+            match Pin::new(&mut self.inner).poll_complete(cx) {
+                Poll::Ready(r) => match r {
+                    Ok(_) => match Pin::new(&mut self.inner).start_seek(pos.into()) {
```

Comment:
I don't think the logic here is correct.  You `poll_complete`, then `start_seek`, then `poll_complete`, which can return Pending. If it does, `poll_fn` will call the closure again when woken, which will `poll_complete`, and then `start_seek` **again**.

Reply:
Not sure what I was thinking before, but LMK if this makes more sense now.